adapter.go

```diff
@@ -182,6 +198,37 @@ func NewAdapterByDBUseTableName(db *gorm.DB, prefix string, tableName string) (*
 	return a, nil
 }
 
+/*
+multiple databases support
+Example usage:
+	dbPool,err := initDbResolver([]gorm.Dialector{mysql.Open(dsn),mysql.Open(dsn2)},[]string{"casbin1","casbin2"})
+	a := initAdapterWithGormInstanceByMulDb(t,dbPool,"casbin1","","casbin_rule1")
+	a = initAdapterWithGormInstanceByMulDb(t,dbPool,"casbin2","","casbin_rule2")
+*/
+func initDbResolver(dbArr []gorm.Dialector, dbNames []string) (DbPool, error) {
+	if len(dbArr) == 0 {
+		panic("dbArr len is 0")
+	}
+	source, e := gorm.Open(dbArr[0])
+	if e != nil {
+		panic(e.Error())
+	}
+	p := specificPolicy{connnIndex: 0}
+	err := source.Use(dbresolver.Register(dbresolver.Config{Policy: &p, Sources: dbArr}))
+	dbMap := make(map[string]int, 3)
+	for i := 0; i < len(dbNames); i++ {
+		dbMap[dbNames[i]] = i
+	}
+	return DbPool{dbMap: dbMap, policy: &p, db: source}, err
+}
+
+func NewAdapterByMulDb(dbPool DbPool, dbName string, prefix string, tableName string) (*Adapter, error) {
+	//change policy
+	dbPool.policy.connnIndex = dbPool.dbMap[dbName]
+	db := dbPool.db.Clauses(dbresolver.Write)
+	return NewAdapterByDBUseTableName(db, prefix, tableName)
+}
```

Comment:
How to change db source when I want?

Reply:
Determined by dbName

Reply:
I mean how to change it after I have created adapter.